frontend/src/components/custom-tools/tool-ide/ToolIde.jsx

```diff
@@ -58,6 +79,16 @@ function ToolIde() {
   const closeLogsModal = () => {
     setShowLogsModal(false);
   };
+  useEffect(() => {
+    if (!shareId && openShareModal) {
+      setOpenShareConfirmation(true);
+      setOpenShareLink(false);
+    }
+    if (shareId && openShareModal) {
+      setOpenShareConfirmation(false);
+      setOpenShareLink(true);
+    }
+  }, [shareId, openShareModal]);
```

Comment:
we first check if openShareModal is true and then handle the states based on shareId. This makes the code cleaner and more readable.
```suggestion
useEffect(() => {
  if (openShareModal) {
    if (shareId) {
      setOpenShareConfirmation(false);
      setOpenShareLink(true);
    } else {
      setOpenShareConfirmation(true);
      setOpenShareLink(false);
    }
  }
}, [shareId, openShareModal]);
```

Reply:
@jagadeeswaran-zipstack Will this not increase the cognitive complexity of the code?

Reply:
@harini-venkataraman This doesn't increase cognitive complexity. Both has cognitive complexity of 2.